magma/backend/coreir/insert_coreir_wires.py

```diff
@@ -119,6 +119,12 @@ def __call__(self, definition):
             for value in instance.interface.ports.values():
                 if not value.is_output():
                     self._insert_wire(value, definition)
+        for builder in definition._context_._builders:
```

Comment:
why do we do this? can we add a comment note for this?

Reply:
Explained in the PR message, but i'll add a comment.

Reply:
More generally, we may want to reconsider when builders are finalized, and have a more unified API for iterating over instances/builders.  The motivation would be that passes may want to modify builders or even introduce them.  Right now we finalize builder when a definition closes, so I'm not sure it would work if a user opened a circuit then added another builder instance? We may want to finalize builders as a final step before compile (which provides a more open interface for circuit modification before compilation)